### PR TITLE
feat(usage): surface prompt-cache split + billed/effective on usage page (#3106)

### LINKED ISSUE #3106: feat(web,api): surface prompt-cache token split + effective/billed tokens on the usage page
## Problem

The v0.0.5 changelog states the usage page "breaks out the cache-read vs. cache-write token split so you can see what caching is saving." **That surfacing never shipped.** `packages/web/src/app/admin/token-usage/tab.tsx` still shows only Gross / Prompt / Completion / Requests, explicitly labelled "not the billed amount," with a `TODO(#3099)` (tab.tsx:181-189) to add the split once the columns are persisted.

Those columns **are** now persisted — migration 0114 added `token_usage.cache_read_tokens` / `cache_write_tokens` (#3103), and prod is writing nonzero values (verified 2026-06-02). So this is now purely a surfacing task, unblocked.

Note: the TODO cites #3099 and the follow-up is sometimes attributed to #3098, but both are **closed** and were about different bugs (the caching no-op and the model-default mismatch, respectively). This surfacing work was untracked until now.

## What changes

Surface, on the usage page:
- A **cache-read vs cache-write** token breakdown (the split that caching produces).
- An **effective/billed** token figure: gross input − prompt-cache discount (cache reads bill ~0.1×, cache writes ~1.25× per Anthropic ephemeral pricing), so customers see the actual savings, not just gross.

## Key files
- `packages/web/src/app/admin/token-usage/tab.tsx` — StatCards + `Usage by Model` table; remove/replace the gross-only caveat + `TODO(#3099)`.
- `packages/api/src/api/routes/admin-tokens.ts` — summary endpoint; aggregate `cache_read_tokens` / `cache_write_tokens` (and a derived effective/billed total).
- `packages/schemas/src/analytics.ts` — extend the summary schema with the new fields (wire type SSOT).

## Acceptance criteria
- [ ] Usage summary endpoint returns cache-read, cache-write, and effective/billed token totals (+ per-model in `byModel`).
- [ ] Usage page displays the cache split and an effective/billed figure distinct from gross.
- [ ] The "not the billed amount" caveat copy is updated to reflect that billed/effective is now shown.
- [ ] The v0.0.5 changelog claim becomes true (or the changelog is corrected in the meantime — see companion docs fix).

## Dependencies
- Data already persisted (migration 0114 / #3103). No blockers.

### PR DESCRIPTION
**Pull request by msywulak**
Closes #3106.

## Why
`token_usage.cache_read_tokens` / `cache_write_tokens` (migration 0114) already persist nonzero values in prod, but the usage page only showed Gross / Prompt / Completion / Requests. This surfaces the prompt-cache split and a billed/effective figure so the v0.0.5 gateway-caching savings are visible — pure surfacing, no new write path.

## What
**API — `packages/api/src/api/routes/admin-tokens.ts`**
- `/tokens/summary` now `SUM`s `cache_read_tokens` / `cache_write_tokens` in both the period-totals and per-model queries (same org filter).
- Derives an **effective/billed token-equivalent**: fresh input ×1.0, cache read ×0.1, cache write ×1.25, output undiscounted. `prompt_tokens` is the **gross** input total and cache read/write are *subsets* of it — confirmed against the AI SDK (`inputTokens = noCache + cacheRead + cacheWrite`, `@ai-sdk/anthropic` → `ai@6` `asLanguageModelUsage`), so `fresh = prompt − cacheRead − cacheWrite`.
- Factors live in named constants with a sourced comment (migration 0114).

**Schema — `packages/schemas/src/analytics.ts`** (wire SSOT)
- `TokenSummary`: `totalCacheReadTokens`, `totalCacheWriteTokens`, `effectiveTokens`.
- `ModelUsageRow`: `cacheReadTokens`, `cacheWriteTokens`, `effectiveTokens`.
- All additive `.default(0)` so a web bundle parsing a pre-cache-split API response still validates mid rolling-deploy (same pattern as the `byModel` `.default([])`).

**UI — `packages/web/src/app/admin/token-usage/tab.tsx`**
- New **Effective / Billed** StatCard (distinct from Gross, shows % below gross) + **Cache Read** / **Cache Write** cards.
- Per-model table gains Cache Read / Cache Write / Effective columns (Total → Gross for naming parity).
- Rewrites the gross-vs-billed caveat; removes the stale `TODO(#3099)`.
- shadcn `StatCard` primitive; `.map` only (immutable).

## Acceptance criteria
- [x] Summary endpoint returns cache-read, cache-write, effective/billed totals (+ per-model in `byModel`).
- [x] Page shows the cache split + an effective figure separate from gross.
- [x] Caveat copy updated; `TODO(#3099)` removed.

## Tests (TDD red→green)
- `admin-tokens.test.ts`: two new `#3106` cases (summary totals + per-model split/effective) and the `#3098` `byModel` `toEqual` updated for the new fields — **16/16**.
- `analytics.test.ts`: additive-default coverage for the new fields — **17/17**.

## Gates
`/ci` green: lint, type, full `bun run test`, syncpack, template-drift, security-headers, railway-watch, schema-drift, oauth-helper-drift, test-discipline, twenty-resolver, published-symbols. (One transient `@useatlas/mcp` `hosted.test` batch-mode mock-pollution flake on the first full run — unrelated to this diff, which touches no MCP code; clean on re-run.)

## Follow-up (post-merge, per task spec)
Once this lands, restore the v0.0.5 changelog claim to "surfaced on the usage page" (currently being walked back to "records … groundwork"). Not in this PR — the v0.0.5 entry is for an already-released tag and is edited direct-to-main.

---
<a href="https://app.blacksmith.sh/AtlasDevHQ/codesmith/atlas/pr/3107"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1783014076&installation_id=135337507&pr_number=3107&repository=AtlasDevHQ%2Fatlas&return_to=https%3A%2F%2Fgithub.com%2FAtlasDevHQ%2Fatlas%2Fpull%2F3107&signature=322751263fa0b78eb97c588284d6e79a1b680a0c0acf11a47b23cae615091a65"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

## Summary by CodeRabbit

* **New Features**
  * Admin token usage shows Cache Read and Cache Write metrics and an updated Effective/Billed token figure; Usage by Model now displays Gross, Cache Read, Cache Write, and Effective columns.
* **Documentation**
  * Expanded explanatory notes clarifying gross vs effective token accounting and prompt-cache re-pricing.
* **Tests**
  * Added tests for cache-split accounting, edge cases (clamping, negatives, repricing), and org-scoped summaries.

### CODE REVIEW

File: packages/schemas/src/analytics.ts [hunk 1]
```diff
@@ -150,13 +166,23 @@ export const ModelUsageRowSchema = z.object({
...
 
 export const TokenSummarySchema = z.object({
   totalPromptTokens: z.number(),
   totalCompletionTokens: z.number(),
   totalTokens: z.number(),
+  // Additive (#3106): `.default(0)` so a web bundle parsing a pre-cache-split
+  // API response still validates during a rolling deploy.
+  totalCacheReadTokens: z.number().default(0),
+  totalCacheWriteTokens: z.number().default(0),
+  effectiveTokens: z.number().default(0),
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/schemas/src/analytics.ts:187]: Addressed in bbd591cd + 037bf0cf. Both surfaces that consume `effectiveTokens` now guard the deploy-window case: when `effectiveTokens === 0 && totalTokens > 0` (an unambiguous absent-field signal, since a correct response can't yield a zero effective with nonzero gross) the Effective/Billed StatCard and the per-model "Effective" column render `—` ("awaiting cache data") instead of `0`/`~100% below gross`. Kept `—` rather than substituting gross, since showing gross would falsely imply zero cache savings during the rollout.
